pkg/kubelet/kubelet_pods.go

```diff
@@ -536,6 +536,11 @@ func (kl *Kubelet) getServiceEnvVarMap(ns string) (map[string]string, error) {
 	}
 
 	for _, e := range envvars.FromServices(mappedServices) {
+		// If service links are not enabled, we shouldn't add service variables
+		// except those which start from "KUBERNETES_".
+		if !*enableServiceLinks && !strings.HasPrefix(e.Name, "KUBERNETES_") {
```

Comment:
It's rather wasteful to call FromServices and then throw away most of it, especially if we want this to be the default mode eventually.

Can we instead change the loop above to test the flag before adding services to the `serviceMap` ?